matchzoo/preprocessor/cdssm_preprocessor.py

```diff
@@ -38,14 +41,28 @@ class CDSSMPreprocessor(engine.BasePreprocessor, preprocessor.SegmentMixin):
         <class 'matchzoo.datapack.DataPack'>
     """
 
-    def __init__(self, sliding_window: int=3):
+    def __init__(self, window_len: int=3, num_windows: int=5,
+                 pad_value: int=0, pad_mode: str='pre',
+                 truncate_mode: str='pre'):
         """Initialization.
 
-        :param sliding_window: sliding window length.
-        :param remove: user-defined removed tokens.
+        :param window_len: sliding window length.
+        :param num_windows: sliding window number.
+        :param pad_value: filling text with :attr:`pad_value` if
+         text length is smaller than assumed.
+        :param pad_mode: String, `pre` or `post`:
+            pad either before or after each sequence.
+        :param truncate_mode: String, `pre` or `post`:
+            remove values from sequences larger than assumed,
+            either at the beginning or at the end of the sequences.
         """
         self.datapack = None
-        self._window = sliding_window
+        self._window_len = window_len
+        self._num_windows = num_windows
+        self._pad_value = pad_value
+        self._pad_mode = pad_mode
+        self._truncate_mode = truncate_mode
+        self._text_length = num_windows + window_len - 1
```

Comment:
@ZizhenWang what is text length?

Reply:
CDSSM preprocessor output data should have fixed windows number, it can be implemented by fixed input text length. 
A window with **length=3** sliding through a text with **length=7** can output **5 windows**. Considering users concern the output shape more than the input shape, I let the user set output `window number` instead of input `text length`.